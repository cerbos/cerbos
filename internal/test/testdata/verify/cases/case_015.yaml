description: Test suite with globals
