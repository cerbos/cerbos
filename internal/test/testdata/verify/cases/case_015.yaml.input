-- testdata/principals.yaml --
---
principals:
  john:
    id: john
    roles:
      - employee
-- testdata/resources.yaml --
---
resources:
  foo:
    kind: global
    id: XX125

-- suite_test.yaml --
---
name: TestSuite
description: Test globals
options:
  globals:
    environment: test
  now: "2022-08-02T15:00:00Z"
tests:
  - name: With global globals
    input:
      principals:
        - john
      resources:
        - foo
      actions:
        - test
        - testfoo
    expected:
      - principal: john
        resource: foo
        actions:
          test: EFFECT_ALLOW
          testfoo: EFFECT_DENY

  - name: With local globals
    options:
      globals:
        environment: hell
        my_map:
          foo: ["wibble", "wobble"]
      now: "2022-08-03T15:00:00Z"
    input:
      principals:
        - john
      resources:
        - foo
      actions:
        - test
        - testfoo
    expected:
      - principal: john
        resource: foo
        actions:
          test: EFFECT_DENY
          testfoo: EFFECT_ALLOW
