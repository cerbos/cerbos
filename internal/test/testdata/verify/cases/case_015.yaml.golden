{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_PASSED",
        "testsCount": 4,
        "resultCounts": [
          {
            "result": "RESULT_PASSED",
            "count": 4
          }
        ]
      },
      "testCases": [
        {
          "name": "With global globals",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "foo",
                  "actions": [
                    {
                      "name": "test",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "testfoo",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        },
        {
          "name": "With local globals",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "foo",
                  "actions": [
                    {
                      "name": "test",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "testfoo",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ],
      "description": "Test globals"
    }
  ],
  "summary": {
    "overallResult": "RESULT_PASSED",
    "testsCount": 4,
    "resultCounts": [
      {
        "result": "RESULT_PASSED",
        "count": 4
      }
    ]
  }
}
